DailymotionPlayerSDK/EventParser.swift

```diff
@@ -4,48 +4,87 @@
 
 import Foundation
 
+public enum PlayerEvent {
+  
+  case timeEvent(name: String, time: Double)
+  case namedEvent(name : String, data: [String: String]?)
+  case errorEvent(error: PlayerError)
+}
+
+public struct PlayerError: Error {
+  let title: String
+  let code: String
+  let message: String
+}
+
 final class EventParser {
   
   private enum Keys {
     static let event = "event"
-    static let time = "time"
-    static let duration = "duration"
+    
+    enum Time {
+      static let time = "time"
+      static let duration = "duration"
+    }
+    
+    enum Error {
+      static let title = "title"
+      static let code = "code"
+      static let message = "message"
+    }
+  }
+  
+  private enum EventName: String {
+    case durationchange
+    case progress
+    case timeupdate
+    case error
   }
 
   static func parseEvent(from: Any) -> PlayerEvent? {
     guard let message = from as? String else { return nil }
-    let eventAndTime = parseEventAndTime(from: message)
-    let time = parseTime(from: eventAndTime)
-    let data = parseData(from: eventAndTime)
+    
+    let eventAndParameters = parseEventAndParameters(from: message)
 
-    guard let event = eventAndTime[Keys.event] else { return nil }
+    guard let event = eventAndParameters[Keys.event] else { return nil }
     
-    switch (event, time, data) {
-    case (let event, .some(let time), _):
-      return .timeEvent(name: event, time: time)
-    case (let event, .none, .some(let data)):
-      return .namedEvent(name: event, data: data)
-    case (let event, .none, .none):
-      return .namedEvent(name: event, data: nil)
+    switch EventName(rawValue: event) {
+    case .none:
+      return .namedEvent(name: event, data: parseData(from: eventAndParameters))
+    case .some(let eventName):
+      switch eventName {
+      case .durationchange, .progress, .timeupdate:
```

Comment:
You must not rely on a list for this, there are more time based events than these (you have a specific time update for ads for example). We don't use it in our own app but some partners may do.